packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -465,6 +546,75 @@ class NotifyingLogger extends Logger {
   }
 }
 
+/// A running application, started by this daemon.
+class AppInstance {
```

Comment:
Much of that is stored in the `RunAndStayResident` class, which is the common code between the `flutter run` and the daemon mode's `app.start`.